waiter/src/waiter/descriptor.clj

```diff
@@ -232,30 +232,36 @@
     (-> (headers/split-headers (:headers request))
         (sd/merge-service-description-sources kv-store waiter-hostnames service-description-defaults token-defaults)
         (sd/merge-service-description-and-id kv-store service-id-prefix current-request-user metric-group-mappings
-                                             service-description-builder assoc-run-as-user-approved?)
+                                             service-description-builder assoc-run-as-user-approved? true)
         (sd/merge-suspended kv-store))))
 
 (defn descriptor->previous-descriptor
```

Comment:
`descriptor->previous-descriptor` doesn't match what this function is doing anymore.   The goal of this change is to change our fallback descriptors.  We have a function for that called `retrieve-fallback-descriptor`.  Should we change it there instead?  Are we intentionally ignoring `fallback-period-secs` for invalid descriptors?

Reply:
> Are we intentionally ignoring fallback-period-secs for invalid descriptors?

Only the `fallback-period-secs` of the latest version of the token is considered. We ignore values of `fallback-period-secs` from previous versions of the token.

Reply:
Moved the logic to `retrieve-fallback-descriptor`